packages/flutter_tools/lib/src/device.dart

```diff
@@ -336,6 +343,15 @@ abstract class DeviceDiscovery {
   /// Gets a list of diagnostic messages pertaining to issues with any connected
   /// devices (will be an empty list if there are no issues).
   Future<List<String>> getDiagnostics() => Future<List<String>>.value(<String>[]);
+
+  /// The well-known device IDs that the discoverer can produce.
```

Comment:
Maybe "The well-known, hard-coded device IDs that the discoverer can produce." it's defining well-known as "well-known".  See: my confusion last week with `ephemeral`
https://github.com/flutter/flutter/blob/d4be5726389869e7843c02337189aa7e5386ef8c/packages/flutter_tools/lib/src/device.dart#L436-L437

Reply:
// This is a comment.

I updated this so that well known is specified  as hard-coded ids